File: location_service/LocationService.Core/Exception/AppException.cs
namespace LocationService.Core.Exception
{
    using System;
    using LocationService.Core.Domain.Constant;

    [Serializable]
    public class AppException : Exception
    {
        public ErrorCode ErrorCode { get; }

        public AppException(ErrorCode errorCode)
            : base(errorCode.GetMessage())
        {
            ErrorCode = errorCode;
        }

        public AppException(ErrorCode errorCode, string message)
            : base(message)
        {
            ErrorCode = errorCode;
        }

        public AppException(ErrorCode errorCode, string message, Exception innerException)
            : base(message, innerException)
        {
            ErrorCode = errorCode;
        }

        // Constructor needed for serialization
        protected AppException(System.Runtime.Serialization.SerializationInfo info,
                             System.Runtime.Serialization.StreamingContext context)
            : base(info, context)
        {
        }
    }
}

File: inventory_service/src/main/java/huy/project/inventory_service/core/domain/exception/AppException.java
package huy.project.inventory_service.core.domain.exception;

import huy.project.inventory_service.core.domain.constant.ErrorCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AppException extends RuntimeException {
    private ErrorCode errorCode;

    public AppException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }
}